src/zampy/datasets/land_cover.py

```diff
@@ -134,16 +135,30 @@ def load(
                 )
                 raise ValueError(msg)
         files = list((ingest_dir / self.name).glob(f"{self.name}_*.nc"))
-
         ds = xr.open_mfdataset(files, chunks={"latitude": 200, "longitude": 200})
         ds = ds.sel(time=slice(time_bounds.start, time_bounds.end))
 
         grid = xarray_regrid.create_regridding_dataset(
             utils.make_grid(spatial_bounds, resolution)
         )
-        ds = ds.regrid.most_common(grid, time_dim="time", max_mem=1e9)
 
-        return ds
+        ds_regrid = {}
+        for variable in variable_names:
+            # select the variable to be regridded
+            da = ds[variable]
+
+            # get values for most common method
+            if "flag_values" in da.attrs:
+                regrid_values = da.attrs["flag_values"]
+            else:
+                regrid_values = np.unique(da.values)
```

Comment:
Do note that this is a very expensive operations, as every single value in the dataarray has to be checked...

It might be better to drop some variables or handle this better, otherwise it won't work on a large area (or be very slow).

The vars `current_pixel_state` and `change_count` have a valid_min and valid_max attr.

`observation_count` does too, but the range is very large, and it is probably irrelevant after regridding (doesn't make physical sense anymore).

Reply:
Good point, I replaced `np.unique` with `da.unique`.